### ISSUE TITLE
SC2086 False negative when command expansion itself contains spaces

### ISSUE THREAD

**Issue by mehrdadn** [koalaman/shellcheck#1736]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2086
- My shellcheck version (`shellcheck --version` or "online"): version: 0.7.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
cmd="/my/folder with/spaces"
$cmd hi
```

#### Here's what shellcheck currently says:

    No issues detected!

#### Here's what I wanted or expected to see:
    Line 3:
    $cmd hi
    ^-- SC2086: Double quote to prevent globbing and word splitting.

**Comment by koalaman** [comment]
You're obviously right in this case.

ShellCheck explicitly suppresses the warning when the variable is used as a command name (since 66cebe7c7) , because the more common reason for having spaces in the command name is flags, e.g. `cmd="git -c core.autocrlf=false"`.

These uses should preferably be rewritten as functions or arrays, but it's still common enough that warning about this case would probably be more noisy than helpful.

**Comment by mehrdadn** [comment]
Is there any workaround for those of us who find it more helpful than noisy? Could you make it opt-in?

I'm trying to detect this bug in a bunch of scripts and finding it impossible because of this.
